lib/service_provider.dart

```diff
@@ -62,11 +68,39 @@ class ServiceProvider {
             localDataRepository: getIt.get(),
           ));
 
+    GetIt.instance.registerFactory<MainPageCubit>(
+      () => MainPageCubit(
+        GetIt.instance.get(),
+      ),
+    );
+
+    GetIt.instance.registerFactory<SettingsCubit>(
+      () => SettingsCubit(
+        GetIt.instance.get(),
+        GetIt.instance.get(),
+      ),
+    );
+
     ///Game Resources
     getIt.registerSingleton<AssetsLoader>(
       AssetsLoader(images: Flame.images..prefix = ''),
     );
+
+    GetIt.instance
+        .registerLazySingleton<SettingsController>(SettingsController.new);
+
+    GetIt.instance
+        .registerLazySingleton<CreatePlayerFunc>(() => AudioPlayer.new);
+
+    GetIt.instance.registerLazySingleton<MusicService>(
+      () => MusicService(
+        GetIt.instance.get(),
+        GetIt.instance.get(),
+      ),
+    );
   }
 
   static T get<T extends Object>() => GetIt.instance.get<T>();
 }
+
+typedef CreatePlayerFunc = AudioPlayer Function();
```

Comment:
Is the Func is a musical style, I don't see a reason why do we need to shorten the word Function, do you?

Reply:
Done

Reply:
Renamed to CreatePlayerFunction